tests/core/test_playback.py

```diff
@@ -789,3 +808,65 @@ def test(self):
         c.playback.pause()
         c.playback.next()
         b.playback.change_track.assert_called_once_with(track2)
+
+
+class Bug1358RegressionTest(unittest.TestCase):
+
+    def setUp(self):  # noqa: N802
+        config = {
+            'core': {
+                'max_tracklist_length': 10000,
+            }
+        }
+
+        self.backend1 = mock.Mock()
+        self.backend1.uri_schemes.get.return_value = ['dummy1']
+        self.playback1 = mock.Mock(spec=backend.PlaybackProvider)
+        self.backend1.playback.change_track.return_value.get.return_value = \
+            False
+        self.backend1.playback = self.playback1
+
+        self.backend2 = mock.Mock()
+        self.backend2.uri_schemes.get.return_value = ['dummy2']
+        self.playback2 = mock.Mock(spec=backend.PlaybackProvider)
+        self.backend1.playback.change_track.return_value.get.return_value = \
+            False
+        self.backend2.playback = self.playback2
+
+        self.tracks = [
+            Track(uri='dummy1:a', length=40000),
+            Track(uri='dummy2:a', length=40000),
+        ]
+
+        self.uris = [
+            'dummy1:a', 'dummy2:a']
```

Comment:
Doesn't need to wrap this. Could also be rewritten as `[t.uri for t in self.tracks]`